File: Toggle_String_Case/Toggle.ts
/*
    Author : Pakshal Shashikant Jain 
    Date : 13/08/2023
    Program : Write a program which accept string from user and toggle the case.
              
              Input : "Marvellous Multi OS"
              Output : mARVELLOUS mULTI os
*/

//Class Declaration
class Toggle
{
    //Characteristisc of Class
    str : string;
    arr : any[];

    //Constructor of Class
    constructor()
    {
        this.str = " ";
        this.arr = [];
    }

    //Behaviour of Class to Take Input from user
    Accept()
    {
        this.str = "Marvellous Multi OS";
        this.arr = Array.from(this.str);
    }

    //Behaviour of Class to Toggle Case 
    Convert()   
    {
        var i : number = 0;
        var ascii : number = 0;

        //Logic To Convert 
        while(i != this.arr.length)
        {
            if((this.arr[i] >= 'a')&&(this.arr[i] <= 'z'))
            {
                ascii = this.arr[i].charCodeAt() - 32;
                
                this.arr[i] = String.fromCharCode(ascii);
            }
            else if((this.arr[i] >= 'A')&&(this.arr[i] <= 'Z'))
            {
                ascii = this.arr[i].charCodeAt() + 32;
                
                this.arr[i] = String.fromCharCode(ascii);
            }

            i++;
        }
    }
}

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
function main() 
{
    console.log("Jay Ganesh......");

    //Creating object of Toggle  Class
    var tobj = new Toggle();

    //Call To Behaviour of Class
    tobj.Accept();
    tobj.Convert();

    //Printing Modified String 
    console.log("Modified String is :",tobj.arr.join(""));
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Call To Main Function
main();

File: Toggle_String_Case/Toggle.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 13/08/2023
    Program : Write a program which accept string from user and toggle the case.
              
              Input : "Marvellous Multi OS"
              Output : mARVELLOUS mULTI os
"""

#####################################################################################################################################################3333

class Toggle :
    #Characteristics of Class    
    arr = [];
    str = " ";

    #Constructor of Class
    def __init__(self) :
        self.arr = [];
        self.str = " ";
    
    #Behaviour of Class To Take Input from user
    def Accept(self) :
        #Taking input from user
        print("Enter String");
        self.str = input();

        self.arr = list(self.str);
    
    #Behaviour of Class To Modify String 
    def Modify(self) :
        i = 0;
        ascii = 0;

        #Logic To Modify String 
        while(i != len(self.arr)) :
            if((self.arr[i] >= 'A')and(self.arr[i] <= 'Z')) :
                ascii = ord(self.arr[i]) + 32;

                self.arr[i] = chr(ascii);
            elif((self.arr[i] >= 'a')and(self.arr[i] <= 'z')) :
                ascii = ord(self.arr[i]) - 32;

                self.arr[i] = chr(ascii);
            i = i + 1;

#######################################################################################################################################################################

#Main Function
def main() :
    print("Jay Ganesh.....");

    #Creating object of Toggle Class
    tobj = Toggle();

    #Call To Behaviour of Class 
    tobj.Accept();
    tobj.Modify();

    #Printing Modified
    print("Modified String is : ","".join(tobj.arr));

################################################################################################################################################

#Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();